### PR TITLE
fix: stop HTML markup leaking into email draft bodies

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- **Root cause:** `ceo-inbox-draft-reply/skill.json` described the `body` field as "HTML body", causing the LLM to write HTML directly. The handler then passes the body through `markdownToHtml()`, which HTML-escapes all tags — so `<p>` becomes `&lt;p&gt;`. Gmail renders the escaped HTML and shows literal `<p>` text in the compose window.
- **Why intermittent:** The LLM follows the manifest faithfully but not always (sometimes writes markdown anyway), so the double-escaping only fires on HTML outputs.
- **Fix:** Two-pronged — correct the manifest description so the LLM writes markdown going forward, and add a `looksLikeHtml()` guard in all three draft handlers to detect HTML input and pass it through unchanged instead of double-escaping it.

## Changes

- `skills/ceo-inbox-draft-reply/skill.json` — corrected `body` description from "HTML body" to "plain text or markdown, never HTML; converted to HTML automatically"; bumped version `0.2.0` → `0.2.1`
- `src/channels/email/markdown-to-html.ts` — added exported `looksLikeHtml()` guard function (detects block-level HTML tags including `html`/`head`/`body` for full-document inputs)
- `skills/ceo-inbox-draft-reply/handler.ts` — apply `looksLikeHtml()` guard before calling `markdownToHtml()`; log.debug when guard fires
- `skills/ceo-inbox-draft-compose/handler.ts` — same guard; moved `htmlBody` computation inside try-catch (was outside, risking uncaught exception)
- `skills/ceo-inbox-draft-edit/handler.ts` — same guard inside new narrow try-catch within the `hasBody` block (was uncaught before)
- `tests/unit/channels/email/markdown-to-html.test.ts` — 11 new test cases for `looksLikeHtml` including full-document HTML coverage

## Test plan

- [ ] `pnpm -C <worktree> test tests/unit/channels/email/markdown-to-html.test.ts` — all 27 cases pass
- [ ] Full test suite passes
- [ ] Typecheck clean
- [ ] Manually trigger a `ceo-inbox-draft-reply` in prod and confirm the draft body renders correctly in Gmail (no visible `<p>` tags)

### CODE REVIEW

**Comment by josephfung** [comment]
@coderabbitai review

File: src/channels/email/markdown-to-html.ts [hunk 1]
```diff
@@ -102,3 +102,22 @@ function escapeHtml(text: string): string {
...
+ * a body field that expects markdown. Passing HTML through markdownToHtml()
+ * would HTML-escape the tags (turning <p> into &lt;p&gt;), which then renders
+ * as visible "<p>" text in the email client.
+ *
+ * Handlers should use this to skip markdownToHtml() when the input is already
+ * HTML, passing the body through unchanged instead.
+ */
+export function looksLikeHtml(text: string): boolean {
+  return /<(p|div|br|ul|ol|li|h[1-6]|blockquote|table|tr|td|th)\b/i.test(text.trim());
+}
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/channels/email/markdown-to-html.ts:123]: Fixed — expanded the regex to include `html`, `head`, and `body` so full-document HTML inputs are caught by the guard. Also added a test case covering `<html><body>...</body></html>` and `<html lang="en"><head>...</head><body>...</body></html>`.

File: tests/unit/channels/email/markdown-to-html.test.ts [hunk 2]
```diff
@@ -130,3 +130,50 @@ describe('markdownToHtml', () => {
...
+    expect(looksLikeHtml('This is <b>important</b>.')).toBe(false);
+  });
+
+  it('returns false for prose that mentions HTML tags as text', () => {
+    // Plain-text discussion of HTML — the guard fires on the tag pattern, so this
+    // is a documented false positive. The trade-off: if an LLM body genuinely
+    // contains "<p>" in prose (e.g. "add a <p> wrapper"), the body is passed through
+    // unchanged rather than being markdown-converted. This is an acceptable edge case.
+    expect(looksLikeHtml('You should use a <p> tag to wrap paragraphs.')).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (josephfung) [reply tests/unit/channels/email/markdown-to-html.test.ts]: Fixed — renamed the test to "returns true for prose that mentions HTML block tags (documented false positive)" which matches the `toBe(true)` assertion. The comment in the test body also clarifies why a false positive here is benign (the passed-through HTML body is already valid) while a false negative would reintroduce the escaped-tag bug.

**Comment by josephfung** [comment]
**CHANGELOG nitpick:** condensed the entry from ~80 words to ~15 per the project guideline. New text: *"HTML tags no longer leak into draft bodies when the LLM writes HTML directly; `looksLikeHtml()` guard added across all draft handlers."*

**Linked issue mismatch warning:** the linked `#1086` was wrong — that's the drift-detector issue. There's no separate GitHub issue tracking this specific HTML-leak bug; removing the `Closes` reference from the PR body.

File: skills/ceo-inbox-draft-compose/handler.ts [hunk 3]
```diff
@@ -89,15 +89,15 @@ export class CeoInboxDraftComposeHandler implements SkillHandler {
...
-      ctx.log.error({ err, subject }, 'ceo-inbox-draft-compose: failed to convert body to HTML');
-      return { success: false, error: 'Failed to convert email body to HTML' };
-    }
+      // Guard: if the LLM wrote HTML directly, pass it through unchanged — running HTML
+      // through markdownToHtml() would escape the tags, making them visible as literal text.
+      const isHtml = looksLikeHtml(body);
+      if (isHtml) {
+        ctx.log.debug({ bodyLength: body.length, subject }, 'ceo-inbox-draft-compose: LLM wrote HTML body — passing through unchanged');
+      }
+      const htmlBody = isHtml ? body : markdownToHtml(body);
```
[the remark this answers is not in the document]
Comment (josephfung) [reply skills/ceo-inbox-draft-compose/handler.ts:99]: Valid concern, but not blocking this PR. A few notes on severity:

1. **Drafts only** — the body goes into a Gmail draft the CEO reviews before sending, not an auto-sent email. The CEO would see a suspicious tracking pixel or link before it leaves the outbox.
2. **Manifest fix reduces the trigger** — now that the manifest says "plain text or markdown, never HTML", the pass-through path only fires when the LLM actively ignores that instruction. Prompt injection would need to override an explicit directive.
3. **Gmail blocks external image auto-load** in compose/drafts — tracking pixels require a manual click.
4. **The pre-PR surface was not clearly better** — when the manifest said "HTML body", the LLM was routinely writing HTML with the intent it be passed through; the escaping was an accident, not a defense.

Agreed that adding sanitization on the pass-through path is the right long-term fix, but it belongs in its own issue/PR: it needs a sanitizer dependency decision, an allowlist, and coverage across all three handlers. Tracking separately.
